src/components/PostsList.tsx

```diff
@@ -1,85 +1,68 @@
 import React from 'react';
+import { Post } from '../types/Post';
 
-export const PostsList: React.FC = () => (
-  <div data-cy="PostsList">
-    <p className="title">Posts:</p>
+type Props = {
+  posts: Post[] | null;
+  targetPost: Post | null;
+  isOpen: boolean;
+  setTargetPost: (post: Post) => void;
+  setIsOpen: (val: boolean) => void;
+};
 
-    <table className="table is-fullwidth is-striped is-hoverable is-narrow">
-      <thead>
-        <tr className="has-background-link-light">
-          <th>#</th>
-          <th>Title</th>
-          <th> </th>
-        </tr>
-      </thead>
+export const PostsList: React.FC<Props> = ({
+  posts,
+  targetPost,
+  isOpen,
+  setTargetPost,
+  setIsOpen,
+}) => {
+  const hendleClick = (post: Post) => {
+    setTargetPost(post);
+    setIsOpen(!isOpen);
+  };
 
-      <tbody>
-        <tr data-cy="Post">
-          <td data-cy="PostId">17</td>
+  return (
+    <div data-cy="PostsList">
+      <p className="title">Posts:</p>
 
-          <td data-cy="PostTitle">
-            fugit voluptas sed molestias voluptatem provident
-          </td>
+      <table className="table is-fullwidth is-striped is-hoverable is-narrow">
+        <thead>
+          <tr className="has-background-link-light">
+            <th>#</th>
+            <th>Title</th>
+            <th> </th>
+          </tr>
+        </thead>
 
-          <td className="has-text-right is-vcentered">
-            <button
-              type="button"
-              data-cy="PostButton"
-              className="button is-link is-light"
-            >
-              Open
-            </button>
-          </td>
-        </tr>
+        {posts && (
```

Comment:
Redundant check. You have this one 
![image](https://user-images.githubusercontent.com/70009205/210411744-6b9eca5c-2f23-420a-b663-b5bb58175196.png)

Reply:
![image](https://user-images.githubusercontent.com/110824803/210457870-5e934ba3-d0bb-4cec-a218-fa5d0f89718a.png)

Reply:
because you yourself made it possible

![image](https://user-images.githubusercontent.com/70009205/210586763-83176869-5c65-4e3f-958e-e52df9e6c83e.png)

Reply:
Code is really nothing but what you created, not some intelligence that knows everything for you, just libraries help us to be more careful and remind us that we write ourselves and forget over time